test-cloud/test-cloud-ha-cep/src/test/java/org/kie/cloud/hacep/HACEPEvaluateRuleIntegrationTest.java

```diff
@@ -0,0 +1,77 @@
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.concurrent.ThreadLocalRandom;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.kie.cloud.api.DeploymentScenarioBuilderFactory;
+import org.kie.cloud.api.scenario.HACepScenario;
+import org.kie.cloud.openshift.resource.Project;
+import org.kie.cloud.openshift.resource.impl.ProjectImpl;
+import org.kie.cloud.tests.common.AbstractMethodIsolatedCloudIntegrationTest;
+import org.kie.hacep.sample.kjar.StockTickEvent;
+import org.kie.remote.CommonConfig;
+import org.kie.remote.RemoteStreamingKieSession;
+import org.kie.remote.TopicsConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HACEPEvaluateRuleIntegrationTest extends AbstractMethodIsolatedCloudIntegrationTest<HACepScenario> {
+
+    private static final Logger logger = LoggerFactory.getLogger(HACEPEvaluateRuleIntegrationTest.class);
+
+    @Override
+    protected HACepScenario createDeploymentScenario(DeploymentScenarioBuilderFactory deploymentScenarioFactory) {
+        return deploymentScenarioFactory.getHACepScenarioBuilder().build();
+    }
+
+    @Test
+    public void testEvaluateRule() throws InterruptedException {
+        final TopicsConfig topicsConfig = TopicsConfig.getDefaultTopicsConfig();
+        final Properties connectionProperties = deploymentScenario.getKafkaConnectionProperties();
+        connectionProperties.putAll(CommonConfig.getStatic());
+        logger.info("Waiting 120 s before starting of sending messages");
+        Thread.sleep(120000);
+        try (RemoteStreamingKieSession producer = RemoteStreamingKieSession.create(connectionProperties, topicsConfig)) {
+            for (int i = 0; i < 1; i++) {
+                StockTickEvent eventA = new StockTickEvent("RHT",
+                                                           ThreadLocalRandom.current().nextLong(80,
+                                                                                                100));
+                producer.insert(eventA);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Producer error", e);
+        }
+
+        logger.info("Waiting 5 second for rule evaluation");
+        Thread.sleep(5000);
+        try (Project project = new ProjectImpl(deploymentScenario.getNamespace())) {
+            final ConfigMap leadersConfigMap = project.getOpenShift().configMaps()
+                    .withName(HACEPTestsConstants.LEADERS_CONFIG_MAP).get();
+            Assertions.assertThat(leadersConfigMap.getData()).containsKey(HACEPTestsConstants.LEADER_POD_KEY);
+            final String leaderPodName = leadersConfigMap.getData().get(HACEPTestsConstants.LEADER_POD_KEY);
+            final String leaderLog = project.getOpenShift().pods().withName(leaderPodName).getLog();
+            Assertions.assertThat(leaderLog).contains("Price for RHT is");
```

Comment:
Would it be possible to retrieve response from HA CEP using REST? It would be better approach than reading logs.

Reply:
Used better approach in improved tests.